segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/exts3/ExtS3Storage.java

```
@@ -0,0 +1,445 @@
+
+                throw new CompletionException(new StreamSegmentNotExistsException(handle.getSegmentName(), null));
+            }
+
+            int originalLength = length;
+
+            int bytesRead = 0;
+            while (length != 0) {
+                bytesRead = reader.read(buffer, bufferOffset, length);
+                length -= bytesRead;
+                bufferOffset += bytesRead;
+            }
+            LoggerHelpers.traceLeave(log, "read", traceId, bytesRead);
+            return originalLength;
+        }
+    }
+
+    private StreamSegmentInformation syncGetStreamSegmentInfo(String streamSegmentName) {
+        long traceId = LoggerHelpers.traceEnter(log, "getStreamSegmentInfo", streamSegmentName);
+        S3ObjectMetadata result = client.getObjectMetadata(config.getExts3Bucket(),
+                config.getExts3Root() + streamSegmentName);
+
+        AccessControlList acls = client.getObjectAcl(config.getExts3Bucket(), config.getExts3Root() + streamSegmentName);
+
+        boolean canWrite = false;
+        canWrite = acls.getGrants().stream().filter((grant) -> {
+            return grant.getPermission().compareTo(Permission.WRITE) >= 0;
+        }).count() > 0;
+
+        StreamSegmentInformation information = new StreamSegmentInformation(streamSegmentName,
+                result.getContentLength(), !canWrite, false,
+                new ImmutableDate(result.getLastModified().toInstant().toEpochMilli()));
+        LoggerHelpers.traceLeave(log, "getStreamSegmentInfo", traceId, streamSegmentName);
+        return information;
+    }
+
+    private boolean syncExists(String streamSegmentName) {
+        GetObjectResult<InputStream> result = null;
+        try {
+            result = client.getObject(config.getExts3Bucket(), config.getExts3Root() + streamSegmentName);
+        } catch (Exception e) {
+            log.warn("Exception {} observed while getting segment info {}", e.getMessage(), streamSegmentName);
```

I don't feel comfortable with swallowing this exception. Let it bubble up and don't record it.